fpdf/fpdf.py

```diff
@@ -2406,98 +2406,81 @@ def _render_styled_cell_text(
-                    self.underline = frag.underline
-                    s_width += self.get_string_width(
-                        frag.string, True
-                    ) + word_spacing * frag.string.count(" ")
-                    s += "] TJ"
+                    sw = self.get_normalized_string_width_with_style(
+                        frag.string, current_font_style
+                    )
+                    sw *= self.font_stretching * self.font_size * 0.00001  # /100 /1000
+                    s_width += sw + self.ws * frag.string.count(" ")
             else:
                 if word_spacing and word_spacing != self.ws:
-                    self._out(f"{word_spacing * self.k:.3f} Tw")
+                    sl.append(f"{word_spacing * self.k:.3f} Tw")
                 elif self.ws > 0:
-                    self._out("0 Tw")
+                    sl.append("0 Tw")
                 self.ws = word_spacing
 
                 for frag in text_line.fragments:
-                    if self.font_style != frag.style:
-                        self.font_style = frag.style
-                        self.current_font = self.fonts[
-                            self.font_family + self.font_style
-                        ]
-                        s += f" /F{self.current_font['i']} {self.font_size_pt:.2f} Tf"
-                        style_changed = True
+                    if current_font_style != frag.style:
+                        current_font_style = frag.style
+                        current_font = self.fonts[self.font_family + current_font_style]
+                        sl.append(f"/F{current_font['i']} {self.font_size_pt:.2f} Tf")
                     if self.unifontsubset:
                         txt_frag_mapped = ""
                         for char in frag.string:
                             uni = ord(char)
-                            txt_frag_mapped += chr(
-                                self.current_font["subset"].pick(uni)
-                            )
-
+                            txt_frag_mapped += chr(current_font["subset"].pick(uni))
                         txt_frag_escaped = escape_parens(
                             txt_frag_mapped.encode("utf-16-be").decode("latin-1")
                         )
                     else:
                         txt_frag_escaped = escape_parens(frag.string)
-                    s += f" ({txt_frag_escaped}) Tj"
+                    sl.append(f"({txt_frag_escaped}) Tj")
                     if frag.underline:
                         underlines.append((self.x + dx + s_width, frag.string))
-                    self.underline = frag.underline
-                    s_width += self.get_string_width(
-                        frag.string, True
-                    ) + self.ws * frag.string.count(" ")
-            s += " ET"
-            # Restoring font style & underline mode after handling changes
-            # by Markdown annotations:
-            if not self._markdown_leak_end_style:
```

Comment:
Seems that you got rid of `self._markdown_leak_end_style`: great, one less hacky attribute!
Could you please get rid of all references to this?

Reply:
Huh, thought I had done that. Now for real.